src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceDataRegistryKey.cs

```diff
@@ -0,0 +1,142 @@
+        {
+            // connect to the specified remote registry
+            SafeRegistryHandle foreignHKey = null;
+            int ret = Interop.Advapi32.RegConnectRegistry(machineName, new SafeRegistryHandle(new IntPtr(PerformanceData), ownsHandle: false), out foreignHKey);
+
+            if (ret == Interop.Errors.ERROR_DLL_INIT_FAILED)
+            {
+                // return value indicates an error occurred
+                throw new ArgumentException(SR.Format(SR.Arg_DllInitFailure, machineName));
+            }
+
+            if (ret != 0)
+            {
+                Win32Error(ret, null);
+            }
+
+            if (foreignHKey.IsInvalid)
+            {
+                // return value indicates an error occurred
+                throw new ArgumentException(SR.Format(SR.Arg_RegKeyNoRemoteConnect, machineName));
+            }
+
+            return new PerformanceDataRegistryKey(foreignHKey);
+        }
+
+        public static PerformanceDataRegistryKey OpenLocal()
+        {
+            var key = new SafeRegistryHandle(new IntPtr(PerformanceData), ownsHandle: true);
+            return new PerformanceDataRegistryKey(key);
+        }
+
+        public byte[] GetValue(string name, bool usePool)
+        {
+            int size = 65000;
+            int sizeInput = size;
+
+            int ret;
+            int type = 0;
+            byte[] data = CreateBlob(size, usePool);
+            while (Interop.Errors.ERROR_MORE_DATA == (ret = Interop.Advapi32.RegQueryValueEx(_hkey, name, lpReserved: null, ref type, data, ref sizeInput)))
+            {
+                if (size == int.MaxValue)
+                {
+                    ReleaseData(data, usePool);
+
+                    // ERROR_MORE_DATA was returned however we cannot increase the buffer size beyond Int32.MaxValue
+                    Win32Error(ret, name);
+                }
+                else if (size > (int.MaxValue / 2))
+                {
+                    // at this point in the loop "size * 2" would cause an overflow
+                    size = int.MaxValue;
+                }
+                else
+                {
+                    size *= 2;
+                }
+                sizeInput = size;
+
+                ReleaseData(data, usePool);
+                data = CreateBlob(size, usePool);
+            }
+
+            if (ret != 0)
+            {
+                ReleaseData(data, usePool);
+                Win32Error(ret, name);
+            }
+
+            return data;
+        }
+
+        public void ReleaseData(byte[] data, bool usePool = true)
+        {
+            if (usePool)
+            {
+#if DEBUG
+                // Fill array by garbage to detect early situations when this array is used after return
+                MemoryMarshal.Cast<byte, uint>(data.AsSpan()).Fill(0xBAADF00D);
+#endif
+
+                ArrayPool<byte>.Shared.Return(data);
+            }
+        }
+
+        public void Close()
```

Comment:
This method is dead code.

Reply:
Sorry?

Reply:
I do not think anybody calls this method and it is internal. Or is this called from somewhere?

Reply:
It is called when `PerformanceCategory` created/renamed or inside of `PerformanceCounter.CloseSharedResources()` call.

Reply:
ok, I have missed that.